### ISSUE TITLE
React-Native AsyncStorage -- Deprecated, Moved to React-Native-Community

### ISSUE THREAD

**Issue by fbartho** [Microsoft/reactxp#1053]
Not sure if this is the right time to flag this as an issue, but React-Native 59 just dropped.

As of this version, accessing `AsyncStorage` from React-Native is Deprecated, and instead we'll need to transition to using a different provider: https://github.com/react-native-community/react-native-async-storage

Used in: https://github.com/Microsoft/reactxp/blob/master/src/native-common/Storage.ts

Other Deprecated/Moved things used by ReactXP:
- NetInfo: 
https://github.com/Microsoft/reactxp/blob/c4a0998319654bc1a28f96028af5db1dfc09af69/src/native-common/Network.ts now moved to: 
https://github.com/react-native-community/react-native-netinfo

Should I make a second ticket for that one?

**Comment by luisnaranjo733** [comment]
Worth noting that the [PR ](https://github.com/Microsoft/reactxp/pull/1059)that references this issue is a step in the right direction, but won't solve the migrations in 0.59 from lean-core as far as I can tell.

I think XP needs to do something like this

```
npm install --save @react-native-community/netinfo
```

And change
```
import { NetInfo } from "react-native";
```
to
```
import NetInfo from "@react-native-community/netinfo";
```

This will be a breaking change to consumer of XP.Network, but it should be backwards compatible with previous versions of RN. Users just now will have an extra step of manually linking the native code from @react-native-community/netinfo which will need

**Comment by berickson1** [comment]
If linking is needed, we may want to consider breaking async storage out into its own extension (reactxp-asyncstorage). This way the reactxp core doesn't have any additional native code dependencies (besides react native core)

**Comment by erictraut** [comment]
Yeah, I like that idea. But it would be a major breaking change, probably requiring us to bump the version to reactxp 2.0

**Comment by linjson** [comment]
@erictraut when do you publish reactxp 2.0?

**Comment by erictraut** [comment]
I don't have a specific timeline in mind. Other than the AsyncStorage change, there haven't been many changes since the last release. Are you in need of this change?

**Comment by mikehardy** [comment]
The lean-core items from react-native will generate a series of breaking changes for reactxp as modules are broken out one by one, even if planned carefully, but they can perhaps be batched.

I just wanted to mention that for RN0.60 (which they are about to cut and put on npmjs tagged as `@next` I think today, WebView and Geolocation modules are completely removed - not just deprecated.  I don't think AsyncStorage (this issue) is gone completely yet but if WebView and Geolocation aren't handled in 1.x then without something up tagged as `@next` for reactxp it will be harder to feel around for where the rough edges are
